controllers/users.js

```diff
@@ -410,13 +411,41 @@ const updateSelf = async (req, res) => {
       }
     }
 
+    if (req.body.disabledRoles) {
+      const data = req.body.disabledRoles;
+      if (user.disabled_roles !== undefined) {
+        rolesToDisable = user.disabled_roles;
+
+        data.forEach((role) => {
+          const roleIndex = rolesToDisable.indexOf(role);
+          if (roleIndex !== -1) {
+            rolesToDisable.splice(roleIndex, 1);
+          } else {
+            rolesToDisable.push(role);
+          }
+        });
+      } else {
+        rolesToDisable = data;
+      }
+    }
+
     if (userRoles.in_discord && !user.incompleteUserDetails) {
       const membersInDiscord = await getDiscordMembers();
       const discordMember = membersInDiscord.find((member) => member.user.id === discordId);
       if (discordMember) {
         const { roles } = discordMember;
         if (roles && roles.includes(discordDeveloperRoleId)) {
-          return res.boom.forbidden("Developers can't update their profile data. Use profile service for updating.");
+          if (req.body.disabledRoles) {
+            const updatedUser = await userQuery.addOrUpdate({ disabled_roles: rolesToDisable }, userId);
+            if (updatedUser) {
+              return res
+                .status(200)
+                .send({ message: "Privilege modified successfully!", disabled_roles: rolesToDisable });
+            }
+          }
+          return res.boom.forbidden(
+            "Developers can only update disabled_roles. Use profile service for updating other attributes."
+          );
```

Comment:
Please create a new Isseus to fix the update flow for users API, and paste the link here

Reply:
https://github.com/Real-Dev-Squad/website-backend/issues/2155